### PR TITLE
[Improvements] Environment changes

### PR DESCRIPTION
**Pull request by shah21**
This PR contains the code changes for multi-environment upcoming improvements

### CODE REVIEW

**Review by gsmithun4 — CHANGES_REQUESTED** [review]

File: server/src/services/apps.service.ts [hunk 1]
```diff
@@ -204,6 +204,17 @@ export class AppsService {
     // removing keys with undefined values
     cleanObject(updatableParams);
     return await dbTransactionWrap(async (manager: EntityManager) => {
+      if (updatableParams.currentVersionId) {
+        //check if the app version is eligible for release
+        const result = await manager.query(
+          'SELECT app_environments.id, app_environments.default FROM app_environments INNER JOIN app_versions ON app_versions.current_environment_id = app_environments.id;'
```
[the remark this answers is not in the document]
Comment (shah21) [reply server/src/services/apps.service.ts:210]: No specific reason. should I change to Typeorm one?

**Review by gsmithun4 — APPROVED** [review]

**Comment by ajith-k-v** [comment]
<h2>Cypress Results</h2>
<table><tr><th>Result</th><th>Passed :white_check_mark:</th><th>Failed :x:</th><th>Pending :hand:</th><th>Skipped :leftwards_arrow_with_hook:</th><th>Duration :clock8:</th></tr><tr><td>Failing :red_circle:</td><td>54</td><td>12</td><td>8</td><td>0</td><td>2379.768s</td></tr></table>
<a href="">

**Review by kavinvenkatachalam — CHANGES_REQUESTED** [review]
@shah21 Getting an error while trying to release an app. Attaching the screenshot

image

**Comment by shah21** [comment]
> @shah21 Getting an error while trying to release an app. Attaching the screenshot
> 
> image

@kavinvenkatachalam  Did you run migration ? or reset the db? if not, please do it

**Review by kavinvenkatachalam — APPROVED** [review]

**Pull request merged**